package/boot/uboot-envtools/files/ramips

```diff
@@ -143,6 +143,10 @@ xiaomi,mi-router-cr6608|\
 xiaomi,mi-router-cr6609)
 	ubootenv_add_uci_config "/dev/mtd1" "0x0" "0x10000" "0x20000"
 	;;
+dna,valokuitu-plus-ex400)
+	ubootenv_add_uci_config "/dev/ubi0_0" "0x0" "0x1f000" "0x1f000" "1"
```

Comment:
This can be done with nvmem.

Reply:
Um how? Do you have an example somewhere? :)

Reply:
https://github.com/openwrt/openwrt/pull/14182

Note that you need to handle mac addressing in userspace. The uboot nvmem driver doesn't support variables besides ethaddr.

Reply:
@neheb eth1addr is only introduced in this PR as ethaddr+1, vendor-fw only has ethaddr. If we can go with nvmem, there is no need for eth1addr at all, because we can specify `nvmem-cells = <&macaddr_uboot_ethaddr 1>;` in the DTS, no?
So we would need neither 99_fwenv-store-ethaddr.sh nor the uboot-envtools change if we use NVMEM in DTS to set the MAC addresses.

Reply:
sure. Note that the way it's done in that PR is probably wrong. It needs to be tested.

Other examples: https://github.com/openwrt/openwrt/blob/main/target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6010-xe3-4.dts#L355

I think that's the correct way. Although note that ethaddr is the only accepted name as of now. See https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/drivers/nvmem/u-boot-env.c?h=v6.6.44#n125

Yeah...

```
#nvmem-cell-cells = <0>;
```

should be

```
```

Reply:
@neheb Please correct me if I am wrong but I think we tried something like this back in March (https://github.com/openwrt/openwrt/pull/13798#issuecomment-1992541780). I had a proposal based on nvmem-on-ubi by @dangowrt and u-boot-env layout proposal by @rmilecki  but it did not fly. Then I resorted to setting u-boot variable eth1addr in user space because it was the cleanest way setting the mac for wan interface at the time.

Looking at upstream the u-boot,env does not support UBI yet (https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/devicetree/bindings/nvmem/u-boot,env.yaml#n24).

If there are pending patches somewhere or progress otherwise I am not aware of and I could base this PR on them, please let me know.

Reply:
Won’t work here then. Unfortunate

Reply:
Yes indeed. Maybe we can convert it later on.

Reply:
I recently worked on support for U-Boot env variables in UBI volumes. It's ready & present in Srini's tree:
https://git.kernel.org/pub/scm/linux/kernel/git/srini/nvmem.git/log/?h=for-next

Once it gets accepted by Greg, I'll backport it to OpenWrt.

There is still no support for writing U-Boot env using NVMEM subsystem.

We'll also need some downstream solution for MACs stored in variables other than `ethaddr`. My upstream proposal got rejected.

Reply:
Hello @rmilecki 

I saw your patches land on main recently. Is it all set and good to go now? I assume a kernel module package is still needed or will architectures just enable NVMEM_LAYOUT_U_BOOT_ENV?

Reply:
`printenv` can't detect NVMEM layout yet, so this line is fine & required.